lib/src/main/java/org/esbtools/lightbluenotificationhook/NotificationHook.java

```diff
@@ -182,70 +183,69 @@ private NotificationEntity makeNotificationEntityWithIncludedFields(HookDoc hook
-            }
+            String valueString = postDoc.get(identityPath).asText();
+            entityData.add(new PathAndValue(pathString, valueString));
         }
-        List<String> changedPaths=new ArrayList<>();
-        List<NotificationEntity.PathAndValue> removedPaths=new ArrayList<>();
-        List<String> removedElements=new ArrayList<>();
-        for(DocComparator.Delta<JsonNode> delta:diff.getDelta()) {
-            if( (delta instanceof DocComparator.Move && arrayOrderSignificant) ||
-                !(delta instanceof DocComparator.Move) ) {
-
-                if(delta instanceof DocComparator.Removal) {
-                    JsonNode removedNode=((DocComparator.Removal<JsonNode>)delta).getRemovedNode();
-                    if(removedNode.isContainerNode()) {
-                        removedElements.add(delta.getField().toString());
-                        flatten(delta.getField().toString(),
-                                ((DocComparator.Removal<JsonNode>)delta).getRemovedNode(),
-                                removedPaths);
-                    } else {
-                        removedPaths.add(new NotificationEntity.PathAndValue(delta.getField().toString(),removedNode.asText()));
-                    }
+
+        // Add flattened include doc to entity data
+        JsonDoc includeDoc=includeProjector.project(postDoc,jsonNodeFactory);
+        flatten("", includeDoc.getRoot(), entityData);
+
+        // Add changes to entity data, removed entity data, changed paths, and removed elements
+        for(DocComparator.Delta<JsonNode> delta : diff.getDelta()) {
+            if (delta instanceof DocComparator.Move && arrayOrderSignificant) {
+                String newPath = delta.getField2().toString();
+                JsonNode movedNode = ((DocComparator.Move<JsonNode>) delta).getMovedNode();
+
+                changedPaths.add(newPath);
+                flatten(newPath, movedNode, entityData);
+            } else if (delta instanceof DocComparator.Removal) {
+                JsonNode removedNode=((DocComparator.Removal<JsonNode>)delta).getRemovedNode();
+                String removedPath = delta.getField().toString();
+
+                if(removedNode.isContainerNode()) {
+                    removedElements.add(removedPath);
+                    flatten(removedPath, removedNode, removedEntityData);
                 } else {
-                    if(delta instanceof DocComparator.Move) {
-                        // Add the new path to the changedPaths
-                        changedPaths.add(delta.getField2().toString());
-                    } else {
-                        changedPaths.add(delta.getField().toString());
-                    }
+                    String removedValue = removedNode.asText();
+                    removedEntityData.add(new PathAndValue(removedPath, removedValue));
                 }
+            } else if (delta instanceof DocComparator.Addition && hookDoc.getPreDoc() != null) {
```

Comment:
I had a hard time parsing the meaning of the conditional here. I'm reading as "this change was an update (vs. an insert)"? For me this might be clearer if we move this logic:

```java
NotificationEntity.Operation operation = hookDoc.getPreDoc() == null
            ? NotificationEntity.Operation.insert
            : NotificationEntity.Operation.update;
```

to the top of the method, and then we can say `operation == Operation.update` or something like that instead of `hookDoc.getPredoc() != null`. Just a thought.